streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamFlatTransformTest.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.kstream.TransformerSupplier;
+import org.apache.kafka.streams.kstream.internals.KStreamFlatTransform.KStreamFlatTransformProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class KStreamFlatTransformTest extends EasyMockSupport {
+
+    private Number inputKey;
+    private Number inputValue;
+
+    private Transformer<Number, Number, Iterable<KeyValue<Integer, Integer>>> transformer;
+    private ProcessorContext context;
+
+    private KStreamFlatTransformProcessor<Number, Number, Integer, Integer> processor;
+
+    @Before
+    public void setUp() {
+        inputKey = 1;
+        inputValue = 10;
+        Transformer<Number, Number, Iterable<KeyValue<Integer, Integer>>> tempTransformer = mock(Transformer.class);
+        transformer = tempTransformer;
+        context = strictMock(ProcessorContext.class);
+        processor = new KStreamFlatTransformProcessor<Number, Number, Integer, Integer>(transformer);
+    }
+
+    @Test
+    public void shouldInitialiseFlatTransformProcessor() {
+        transformer.init(context);
+        replayAll();
+
+        processor.init(context);
+
+        verifyAll();
+    }
+
+    @Test
+    public void shouldTransformInputRecordToMultipleOutputRecords() {
+        List<KeyValue<Integer, Integer>> outputRecords = Arrays.asList(
+                KeyValue.pair(2, 20),
+                KeyValue.pair(3, 30),
+                KeyValue.pair(4, 40));
+        processor.init(context);
+        EasyMock.reset(transformer);
+
+        EasyMock.expect(transformer.transform(inputKey, inputValue)).andReturn(outputRecords);
+        context.forward(outputRecords.get(0).key, outputRecords.get(0).value);
+        context.forward(outputRecords.get(1).key, outputRecords.get(1).value);
+        context.forward(outputRecords.get(2).key, outputRecords.get(2).value);
+        replayAll();
+
+        processor.process(inputKey, inputValue);
+
+        verifyAll();
+    }
+
+    @Test
+    public void shouldTransformInputRecordToEmptyList() {
+        processor.init(context);
+        EasyMock.reset(transformer);
+
+        EasyMock.expect(transformer.transform(inputKey, inputValue))
+                            .andReturn(Collections.<KeyValue<Integer, Integer>>emptyList());
+        replayAll();
+
+        processor.process(inputKey, inputValue);
+
+        verifyAll();
+    }
+
+    @Test
+    public void shouldCloseFlatTransformProcessor() {
+        transformer.close();
+        replayAll();
+
+        processor.close();
+
+        verifyAll();
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
Is this save? Test would pass if NPE is thrown from somewhere unexpected... I would prefer to keep the scope where NPE is expected as tide as possible. We often use:
```
try {
  toSomethingAndExpectNPE();
  fail("Should have thrown");
} catch (final NPE expected) { }
```

Not sure if we can rewrite this way. Doesn't EasyMock not allow to expect an exception, too (as another alternative)?